File: trunk/BrushSide.java
// BrushSide class

// This class holds data of a single brush side.

public class BrushSide {
	
	// INITIAL DATA DECLARATION AND DEFINITION OF CONSTANTS
	
	private int plane;
	private int face; // I don't understand why this lump references
	                  // faces. Brushes are referenced by leaves which
							// also references faces. It seems like it's
							// duplicating faces for no good reason.
	
	// CONSTRUCTORS
	
	// This one takes the components separate and in the correct data type
	public BrushSide(int inPlane, int inFace) {
		plane=inPlane;
		face=inFace;
	}

	// This one takes an array of bytes (as if read directly from a file) and reads them
	// directly into the proper data types.
	public BrushSide(byte[] in) throws InvalidBrushSideException {
		if(in.length!=8) {
			throw new InvalidBrushSideException();
		}
		plane=(in[3] << 24) | ((in[2] & 0xff) << 16) | ((in[1] & 0xff) << 8) | (in[0] & 0xff);
		face=(in[7] << 24) | ((in[6] & 0xff) << 16) | ((in[5] & 0xff) << 8) | (in[4] & 0xff);
	}
	
	// METHODS
	
	// ACCESSORS/MUTATORS
	public int getPlane() {
		return plane;
	}
	
	public void setPlane(int in) {
		plane=in;
	}

	public int getFace() {
		return face;
	}
	
	public void setFace(int in) {
		face=in;
	}
}

File: LumpTools/BrushSide.cs
using System;
// BrushSide class

public class BrushSide:LumpObject {
	
	// INITIAL DATA DECLARATION AND DEFINITION OF CONSTANTS
	
	private int plane = - 1;
	private int face = - 1;
	
	// CONSTRUCTORS
	public BrushSide() {
	}

	public BrushSide(LumpObject bobSaget):base(bobSaget.Data) {
		new BrushSide(bobSaget.Data);
	}
	
	public BrushSide(byte[] data):base(data) {
		face = DataReader.readInt(data[0], data[1], data[2], data[3]);
		plane = DataReader.readInt(data[4], data[5], data[6], data[7]);
	}
	
	// METHODS
	public byte[] toByteArray() {
		byte[] ret = new byte[8];
		byte[] temp = BitConverter.GetBytes(face);
		Array.Copy(temp, 0, ret, 0, 4);
		temp = BitConverter.GetBytes(plane);
		Array.Copy(temp, 0, ret, 4, 4);
		return ret;
	}

	public static Lump<BrushSide> createLump(byte[] data) {
		int structLength = 8;
		int offset = 0;
		Lump<BrushSide> lump = new Lump<BrushSide>(data.Length, structLength, data.Length / structLength);
		byte[] bytes = new byte[structLength];
		for (int i = 0; i < data.Length / structLength; i++) {
			for (int j = 0; j < structLength; j++) {
				bytes[j] = data[offset + j];
			}
			lump.Add(new BrushSide(bytes));
			offset += structLength;
		}
		return lump;
	}

	// ACCESSORS/MUTATORS

	virtual public int Plane {
		get {
			return plane;
		}
		set {
			plane = value;
		}
	}

	virtual public int Face {
		get {
			return face;
		}
		set {
			face = value;
		}
	}
}